superset/views/annotations.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from flask_babel import gettext as __
+from flask_appbuilder.models.sqla.interface import SQLAInterface
+
+from superset.models.annotations import Annotation, AnnotationLayer
+from superset import appbuilder
+from .base import SupersetModelView, DeleteMixin
+
+
+class AnnotationModelView(SupersetModelView, DeleteMixin):  # noqa
```

Comment:
How are annotations written to that table? If you cannot write them from within superset, why are you not thinking of them as just another datasource?

Reply:
This feature ships with a CRUD that allows users to manually input/manage layers. FAB also exposes a CURD REST API to maintain those programmatically, though I'm thinking at Airbnb most likely people will write Airflow jobs talking to our MySQL instance to load annotations dynamically.

Reply:
I don't think that this table should be part of the Superset DB then. The way Superset is currently designed is that it is agnostic (as long as there is a Druid/SQL connector) to where the data is stored. We have a clean separation of meta data (slice definition, configuration, users, ...) and the data that superset is operating on.
This table inverses this fundamentally and now we have to store presentation data directly inside of the superset db. Currently no time series data is stored inside of the superset database. I'm also worried that this will cause scalability issues. Now my superset meta database is going to have very different performance requirements with regard to querying time series.

If this table was only used for manually created data from a user interface you could make the argument that we are talking about a very limited amount of data which originated in superset and therefore should be stored there. But this seems not to be the case.

Reply:
Yeah it's at the frontier of data/metadata, though to me annotations are closer to metadata.

Reply:
Though I agree that it would be sweet to support external data for annotations (as in your PR), let's do both. I have some thoughts on that that I'll put on your PR. The TLDR is I think we should create a new `time_annotations` viz_type to enforce a certain set of guaranties, as opposed to a convention on top of the `table` viz_type.